content/en/docs/armory-agent/agent-mtls.md

```diff
@@ -0,0 +1,267 @@
+
+{{< prism lang="yaml" line="20-21, 26-28" >}}
+spec:
+  kustomize:
+    clouddriver:
+      deployment:
+        patchesStrategicMerge:
+          - |
+            spec:
+              template:
+                spec:
+                  initContainers:
+                  - name: kubesvc-plugin
+                    image: docker.io/armory/kubesvc-plugin:<release>
+                    volumeMounts:
+                      - mountPath: /opt/plugin/target
+                        name: kubesvc-plugin-vol
+                  containers:
+                  - name: clouddriver
+                    volumeMounts:
+                      - mountPath: /opt/clouddriver/lib/plugins
+                        name: kubesvc-plugin-vol
+                      - mountPath: /opt/clouddriver/cert  
+                        name: cert
+                  volumes:
+                  - name: kubesvc-plugin-vol
+                    emptyDir: {}
+                  - name: cert
+                    secret:
+                      secretName: <your-secret-name>
+{{< /prism >}}
+
+
+### Configure the plugin
+
+Configure the plugin to use the mounted certs in the `agent-plugin/config.yaml` file. Note that `certificateChain`, `trustCertCollection`, and `privateKey` values must in `file:///filepath/filename` format.
+
+{{< prism lang="yaml" line="12-20" >}}
+apiVersion: spinnaker.armory.io/{{< param "operator-extended-crd-version" >}}
+kind: SpinnakerService
+metadata:
+  name: spinnaker
+spec:
+  spinnakerConfig:
+    profiles:
+      clouddriver:
+        # See https://docs.armory.io/docs/installation/armory-agent/agent-options/
+        kubesvc:
+          cluster: redis
+          grpc:
+            server:
+              security:
+                enabled: true
+                certificateChain: file:<path-to-agent.crt>
+                trustCertCollection: file:<path-to-agent.crt>
+                privateKey: file:<path-to-agent.key>
+                clientAuth: REQUIRE
+{{< /prism >}}
+
+See the {{< linkWithTitle "agent-plugin-options.md" >}} page for additional options.
+
+## Agent service configuration
+
+### Create a secret
+
+Create a secret in the target cluster namespace where the Agent resides.
+
+```bash
+kubectl create secret generic <agent-secret-name> \
+--from-file=agent.crt --from-file=agent.key
+```
+
+### Modify deployment configuration
+
+Modify the Agent's deployment configuration to mount the certs. @TODO what file is this in?
+
+If you have a custom CA, you need to mount the cert into the known trusted cert location:  `/etc/ssl/cert.pem`. Golang apps in Alpine use `/etc/ssl/cert.pem` as the source of a trusted CA.
+
+  1. Obtain a copy of the `cert.pem` file from a running Agent.
+  2. Append your custom CA onto the copied `cert.pem`  (for example `cat ca.crt >> cert.pem`)
+  3. Mount `cert.pem` to Agent deployment
+
+@TODO is this config correct? do the "this didn't work" sections work now or not?
```

Comment:
is this yaml snippet correct?

Reply:
Note: The steps to programmatically do steps 1-3 are not included in this documentation. I'm not sure what those steps are or should be. If the user is going this route, it's probable they know how to do this.